langchain/vectorstores/base.py

```diff
@@ -116,6 +117,14 @@ def similarity_search_with_relevance_scores(
         """Return docs and relevance scores in the range [0, 1].
 
         0 is dissimilar, 1 is most similar.
+
+        Args:
+            query: input text
+            k: Number of Documents to return. Defaults to 4.
+            score_threshold: Optional, a floating point value between 0 to 1 to filter the resulting set of retrieved docs
```

Comment:
can we add this as a method to the retriever itself, rather than the vectorsotre? as the logic added (thresholding) is much more to do with retrieval

Reply:
It's actually an optional argument we can pass to the `VectorStoreRetriever` class that will then pass to the downstream retrieval function `similarity_search_with_relevance_scores`. Before, `VectorStoreRetriever` only supports `similarity`(always returns top k) or `mmr` retrieval method.  Do you mean only apply the filtering logic in the `get_relevant_documents` method of the  `VectorStoreRetriever` class.

Reply:
facing similar issue